src/client/src/Client.c

```diff
@@ -146,6 +226,19 @@ STATUS createKinesisVideoClient(PDeviceInfo pDeviceInfo, PClientCallbacks pClien
 
     pKinesisVideoClient->base.pStateMachine = pStateMachine;
 
+    if(pKinesisVideoClient->deviceInfo.clientInfo.automaticStreamingFlags == AUTOMATIC_STREAMING_INTERMITTENT_PRODUCER) {
+        if (!IS_VALID_TIMER_QUEUE_HANDLE(pKinesisVideoClient->timerQueueHandle)) {

```

Comment:
Not sure why you check for the timer queue handle validity. This will be isolated per client object so when the client is being created, it will be unique.